Tools/ardupilotwaf/boards.py

```diff
@@ -148,9 +148,15 @@ def configure_env(self, cfg, env):
                 '-O0',
             ]
 
-        env.LINKFLAGS += [
-            '-Wl,--gc-sections',
-        ]
+        if sys.platform == 'darwin':
```

Comment:
Please use `cfg.env.DEST_OS` instead of `sys.platform`. If we are cross-compiling in macOS we won't use macOS linker so we can't rely on the platform we are building in.